src/Spice86.Core/Emulator/OperatingSystem/DosFilePathResolver.cs

```diff
@@ -0,0 +1,172 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using Spice86.Shared.Emulator.Errors;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+/// <inheritdoc />
+public class DosFilePathResolver : IDosFilePathResolver {
+    private readonly ILoggerService _loggerService;
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="loggerService">The logger service implementation.</param>
+    public DosFilePathResolver(ILoggerService loggerService) => _loggerService = loggerService;
+    
+    /// <inheritdoc />
+    public string GetHostParentDirectory(string path) => Directory.GetParent(path)?.FullName ?? path;
+
+    /// <inheritdoc />
+    public string? GetActualCaseForFileName(string caseInsensitivePath) {
+        string? directory = Path.GetDirectoryName(caseInsensitivePath);
+        string? directoryCaseSensitive = GetDirectoryCaseSensitive(directory);
+        if (string.IsNullOrWhiteSpace(directoryCaseSensitive) || Directory.Exists(directoryCaseSensitive) == false) {
+            return null;
+        }
+        string realFileName = "";
+        string[] array = Directory.GetFiles(directoryCaseSensitive);
+        foreach (string file in array) {
```

Comment:
Looks like this is already done in GetDirectoryCaseSensitive isnt it?

Reply:
I don't think so. Here, we list files in the folder returned by GetDirectoCaseSensitive.

I renamed the method to TryGetHostFullNameForFile and removed a useless check.